main.go

```diff
@@ -1,152 +1,105 @@
 package main
 
 import (
-	"context"
-	"encoding/json"
 	"fmt"
-	"io"
-	"io/ioutil"
 	"net"
 	"os"
 	"os/signal"
 	"syscall"
 
-	log "github.com/sirupsen/logrus"
+	"github.com/hashicorp/go-hclog"
+	providerserver "github.com/hashicorp/secrets-store-csi-driver-provider-vault/internal/server"
+	"github.com/hashicorp/secrets-store-csi-driver-provider-vault/internal/version"
 	"github.com/spf13/pflag"
 	"google.golang.org/grpc"
 	pb "sigs.k8s.io/secrets-store-csi-driver/provider/v1alpha1"
 )
 
 var (
-	endpoint = pflag.String("endpoint", "/tmp/vault.sock", "path to socket on which to listen for driver gRPC calls")
-	debug    = pflag.Bool("debug", false, "sets log to debug level")
-	version  = pflag.Bool("version", false, "prints the version information")
+	endpoint    = pflag.String("endpoint", "/tmp/vault.sock", "path to socket on which to listen for driver gRPC calls")
+	debug       = pflag.Bool("debug", false, "sets log to debug level")
+	selfVersion = pflag.Bool("version", false, "prints the version information")
 )
 
-const minDriverVersion = "v0.0.17"
-
-// LogHook is used to setup custom hooks
-type LogHook struct {
-	Writer    io.Writer
-	Loglevels []log.Level
+func main() {
+	logger := hclog.Default()
+	err := realMain(logger)
+	if err != nil {
+		logger.Error("Error running provider", "err", err)
+		os.Exit(1)
+	}
```

Comment:
nit: HashiCorp projects tend to use the Mitchell's CLI package. This is helpful to process parameters/flags and in general have consistency across the projects. This isn't a huge deal since its exclusively running in a Docker image, but I'd suggest taking a look at Vault K8s: https://github.com/hashicorp/vault-k8s/blob/master/main.go#L12-L21.

Reply:
Will also add this in a future PR 👍